spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/admin/DatabaseUtilityProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core.admin;
+
+/**
+ * A provider of Cloud Spanner database-related objects.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.2
+ */
+public interface DatabaseUtilityProvider<T> {
```

Comment:
This looks identical to `java.util.function.Supplier` except for the name. Why not just use `Supplier` in its place?

Reply:
Yea I actually prefer using those general "functional" classes. Will change.